client/src/context/PageContext.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+
+const PageContext = createContext();
+
+export const PageProvider = ({ children }) => {
+  const [currentPage, setCurrentPage] = useState(
+    sessionStorage.getItem("currentPage") || "dashboard"
```

Comment:
This is just to address the issue of if a user refreshes while on the "profile" page.
Refreshes force a re-render of the app and would reset currentPage to become the default value ("dashboard")